app/src/main/java/org/liberty/android/fantastischmemo/ui/GradeButtonsFragment.java

```diff
@@ -180,13 +188,36 @@ public boolean onTouch(View v, MotionEvent event) {
         return buttonView;
     }
 
+    public ArrayList<Card> getForgotCards() {
+        return forgotCards;
+    }
+
+    public ArrayList<Card> getRememberedCards() {
+        return rememberedCards;
+    }
+
     public void gradeCurrentCard(int grade) {
         onGradeButtonClickListener.onGradeButtonClick(grade);
     }
 
     private void setButtonOnClickListener(final Button button, final int grade) {
+
         button.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
+                if (grade == 0 || grade == 1) {
+                    Parcel parcel = Parcel.obtain();
+                    Card currentCard = activity.getCurrentCard();
+                    forgotCards.add(currentCard);
+                    parcel.writeValue(currentCard);
+                    currentCard.writeToParcel(parcel, 0);
+                }
+                else {
+                    Parcel parcel = Parcel.obtain();
+                    Card currentCard = activity.getCurrentCard();
```

Comment:
Maybe put the parcel and current card declaration above the if (line 207) so that it is not repeated in the if and else